Mytinerary/View Controllers/InputEventViewController.m

```diff
@@ -0,0 +1,294 @@
+    [NSLayoutConstraint activateConstraints:stackConstraints];
+    
+    // store pointers to stackView and scrollView to render additional input forms
+    self.stackView = stackView;
+    self.scrollView = scrollView;
+    
+    // add shared event input view
+    [stackView addArrangedSubview:self.eventInputSharedView];
+    [self.eventInputSharedView.heightAnchor constraintEqualToConstant:EVENT_INPUT_SHARED_VIEW_HEIGHT].active = YES;
+    
+    // by default, add activity event input view
+    [stackView addArrangedSubview:self.eventInputActivityView];
+    [self.eventInputActivityView.heightAnchor constraintEqualToConstant:EVENT_INPUT_ACTIVITY_VIEW_HEIGHT].active = YES;
+    
+    // add submit button view
+    [stackView addArrangedSubview:self.eventInputSubmitView];
+    [self.eventInputSubmitView.heightAnchor constraintEqualToConstant:EVENT_INPUT_SUBMIT_VIEW_HEIGHT].active = YES;
+    
+    // setup shared category picker view
+    self.eventInputSharedView.categoryPickerView.delegate = self;
+    self.eventInputSharedView.categoryPickerView.dataSource = self;
+    self.eventCategoryPickerData = [NSArray arrayWithObjects:@"activity", @"transportation", @"food", @"hotel", nil];
+    
+    // setup transportation category picker view
+    self.eventInputTransportationView.typePickerView.delegate = self;
+    self.eventInputTransportationView.typePickerView.dataSource = self;
+    self.transportationTypePickerData = [NSArray arrayWithObjects:@"walk", @"bike", @"car", @"public transportation", nil];
+    
+    // setup food cost picker view
+    self.eventInputFoodView.costPickerView.delegate = self;
+    self.eventInputFoodView.costPickerView.dataSource = self;
+    self.foodCostPickerData = [NSArray arrayWithObjects:@"$", @"$$", @"$$$", @"$$$$", nil];
+    
+    // setup hotel type picker view
+    self.eventInputHotelView.typePickerView.delegate = self;
+    self.eventInputHotelView.typePickerView.dataSource = self;
+    self.hotelTypePickerData = [NSArray arrayWithObjects:@"hotel", @"campground", @"hostel", @"airbnb", nil];
+}
+
+- (IBAction)onTapCloseButton:(id)sender {
+    [self dismissViewControllerAnimated:YES completion:nil];
+}
+
+// on submitting new event form
+- (IBAction)onTapSubmitButton:(id)sender {
+    int selectedCategoryIdx = (int)[self.eventInputSharedView.categoryPickerView selectedRowInComponent:0];
+    NSString *selectedCategory = self.eventCategoryPickerData[selectedCategoryIdx];
+    
+    // initialize event depending on selected event type
+    if ([selectedCategory isEqualToString:@"activity"]) {
+        [self initActivityEvent];
+    }
+    else if ([selectedCategory isEqualToString:@"transportation"]) {
+        [self initTransportationEvent];
+    }
+    else if ([selectedCategory isEqualToString:@"food"]) {
+        [self initFoodEvent];
+    }
+    else if ([selectedCategory isEqualToString:@"hotel"]) {
+        [self initHotelEvent];
+    }
```

Comment:
might be able to use a switch statement to make code sections like these a little cleaner

Reply:
I think with string matching you have to use if/else chains because of string comparison (rather than strict equality). I could be wrong though